kyuubi-server/src/main/scala/org/apache/kyuubi/engine/spark/SparkProcessBuilder.scala

```diff
@@ -102,6 +105,27 @@ class SparkProcessBuilder(
     }
   }
 
+  override protected def engineScalaBinaryVersion: String =
+    StringUtils.defaultIfBlank(System.getenv("SPARK_SCALA_VERSION"), sparkCoreScalaVersion)
+
+  override lazy val homeDirFilter: FilenameFilter = {
+    if (SCALA_COMPILE_VERSION.equals("2.13")) {
+      new RegexFileFilter(
+        "^spark-(\\d+\\.\\d+\\.\\d+)-bin-hadoop\\d+-scala\\d+\\.\\d+$",
+        IOCase.INSENSITIVE)
+    } else {
+      new RegexFileFilter("^spark-\\d+\\.\\d+\\.\\d+-bin-hadoop\\d+$", IOCase.INSENSITIVE)
```

Comment:
the hadoop pattern is incorrect

```
spark-3.1.3-bin-hadoop3.2.tgz
spark-3.4.1-bin-hadoop3.tgz
```

Reply:
INSENSITIVE seems is not required

Reply:
Also `dir.isDirectory` should be applied too

Reply:
I have fixed the regex pattern according to your detailed suggestions, thanks.
The `dir.isDirectory` could be skipped, as the dir in FileFilter is guaranteed to be a folder directory.

Reply:
After the second thought, it's kept to check `isDirectory` but fixed to checks on file itself by using `FileFilter` rather than `FileNameFileter`.